client/engine/messageservice/test-messageservice.go

```diff
@@ -115,7 +115,6 @@ func (tms TestMessageService) routeFromPeers() {
 // Close stops the TestMessagerService from sending or receiving messages.
 func (tms TestMessageService) Close() error {
 	close(tms.fromPeers)
-	close(tms.out)
```

Comment:
I'm not sure I agree that 

> There is no goroutine waiting for the output of the channel, so
closing the channel is not necessary.

The engine run loop is one such goroutine. It does alias the name of the channel, so that may explain the confusion. 

That goroutine is likely being cleaned up by another mechanism. But it's not unlikely that we refactor things such that there is a dedicated goroutine reading from this channel.

Reply:
I am open to adding this back with the caveat of https://github.com/statechannels/go-nitro/pull/1174#issuecomment-1481420768. We should settle on a pattern and stick with it. In this PR I recommend one pattern (only close channels when necessary), but I am open to alternative patterns.